src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "debug_bundle_service.h"
+
+#include "config/configuration.h"
+#include "debug_bundle/error.h"
+
+#include <seastar/core/seastar.hh>
+#include <seastar/core/shard_id.hh>
+
+namespace debug_bundle {
+static ss::logger lg{"debug-bundle-service"};
+
+debug_bundle_service::debug_bundle_service(
+  const std::filesystem::path& data_dir)
+  : _debug_bundle_dir(data_dir / debug_bundle_dir_name)
+  , _rpk_path_binding(config::shard_local_cfg().rpk_path.bind())
+  , _rpk_path(config::shard_local_cfg().rpk_path()) {
+    _rpk_path_binding.watch([this] {
+        _rpk_path = config::shard_local_cfg().rpk_path();
+        lg.trace("Changing rpk path to {}", _rpk_path);
```

Comment:
I don't think `_rpk_path` adds anything over using `operator()()` on the binding?